enhancements/machine-api/cluster-autoscaler-operator.md

```diff
@@ -0,0 +1,290 @@
+as a Deployment with opinionated default options.
+
+Provide a way for users to opt individual MachineSets into autoscaling and to set boundaries for the size of the
+MachineSets that they wish to autoscale.
+
+## Motivation
+
+### Goals
+
+- Enable users to create Cluster Autoscaler with opinionated defaults
+- Enable users to autoscale MachineSets
+- Enable users to set a maximum size of their cluster
+- Enable users to set maximum and minimum sizes for each of their MachineSets
+
+### Non-Goals
+
+- Integrate the Cluster Autoscaler with Machine API (This will be a separate proposal)
+- Enable autoscaling by default for new OpenShift clusters
+- Provide options for autoscaling across multiple MachineSets
+
+## Proposal
+
+A new controller will be developed to manage the lifecycle of the Cluster Autoscaler application and to manage
+assigning size limits to MachineSets to enable the Cluster Autoscaler to autoscale desired MachineSets.
+
+### User Stories
+
+#### Story 1
+
+As a user of OpenShift, it should be easy to enable autoscaling of compute resources so that my cluster will respond to
+increased usage without manual intervention.
+
+#### Story 2
+
+As a user of OpenShift, I should be able to specify which groups of Machines should be autoscaled and which should not.
+
+#### Story 3
+
+As a user of OpenShift, I should be able to limit the size of my cluster when it is autoscaling so that I do not overload
+the Control Plane and so that I can cap the costs of my cluster.
+
+#### Story 4
+
+As a user of OpenShift, I should be able to specify a maximum and minimum size for any group of Machines being scaled
+so that I can ensure I have a minimum capacity available and that I do not over commit to a single pool of Machines.
+
+### Implementation Details
+
+The controller will consist of two control loops, one for a new `ClusterAutoscaler` CRD, and the second for a new
+`MachineAutoscaler` CRD.
+
+This controller will be deployed and managed by the Cluster Version Operator and will become a second level operator.
+
+#### ClusterAutoscaler controller
+
+The `ClusterAutoscaler` controller is responsible for managing Cluster Autoscaler Deployments in the `openshift-machine-api` namespace.
+It will also be responsible for ensuring that the relevant resources for providing monitoring information will also be deployed.
+
+If a `ClusterAutoscaler` resource exists, the controller will perform the following logic:
+- Fetch the `ClusterAutoscaler` custom resource
+- Validate the options specified on the `ClusterAutoscaler` specification
+  - Should the validation fail, log an error and requeue the custom resource
+- Check for a Deployment for the `ClusterAutoscaler` custom resource
+  - A Deployment named `cluster-autoscaler-<name>` would be expected, where `<name>` is the name of the `ClusterAutoscaler` resource
+- Ensure monitoring resources are set up for the Cluster Autoscaler deployment
```

Comment:
Include basic telemeter backhaul here:  # machinesets, size of each, instance type, autoscaler configuration/watermarks

Reply:
Good suggestion, will expand here

Reply:
I added a section detailing the alerts we deploy. Don't think it's worth elaborating on the particular metrics that will be gathered as these come from the Cluster Autoscaler, not the Operator